commit 51e645b6652c997a68cbadf70e62a7b0c49e63a2
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Thu Jun 7 14:45:58 2018 +0100

    drm/i915: Mark the GPU as wedged without error on fault injection
    
    If we have been instructed (by CI) to inject a fault to load the module
    with a wedged GPU, do so quietly less we upset CI.
    
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
    Cc: Michał Winiarski <michal.winiarski@intel.com>
    Cc: Michal Wajdeczko <michal.wajdeczko@intel.com>
    Reviewed-by: Michał Winiarski <michal.winiarski@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20180607134558.31150-1-chris@chris-wilson.co.uk

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index fd882eb389d2..4187e0688e50 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -5491,6 +5491,8 @@ int i915_gem_init(struct drm_i915_private *dev_priv)
 	}
 
 	if (i915_inject_load_failure()) {
+		DRM_DEBUG_DRIVER("Marking the driver as wedged\n");
+		i915_gem_set_wedged(dev_priv);  /* Fail silently! */
 		ret = -EIO;
 		goto err_init_hw;
 	}